tests/unit/maxmemory.tcl

```diff
@@ -210,18 +213,24 @@ proc test_slave_buffers {test_name cmd_count payload_len limit_memory pipeline}
             set used_no_repl [expr {$new_used - $mem_not_counted_for_evict}]
             set delta [expr {($used_no_repl - $client_buf) - ($orig_used_no_repl - $orig_client_buf)}]
 
+            # Calculate query buffer size of slave
+            set clients [split [$master client list] "\r\n"]
+            set c [lsearch -inline $clients *flags=S*]
+            assert {[regexp {qbuf=([0-9]+)} $c - qbuf]}
+            assert {[regexp {qbuf-free=([0-9]+)} $c - qbuf_free]}
+            set slave_qbuf [expr $qbuf + $qbuf_free]
+
             assert {[$master dbsize] == 100}
             assert {$slave_buf > 2*1024*1024} ;# some of the data may have been pushed to the OS buffers
-            set delta_max [expr {$cmd_count / 2 + 40*1024}] ;# 1 byte unaccounted for, with 1M commands will consume some 1MB
-                                                             # querybuf of slave will consume 40k with jemalloc
+            set delta_max [expr {$cmd_count / 2}] ;# 1 byte unaccounted for, with 1M commands will consume some 1MB
             assert {$delta < $delta_max && $delta > -$delta_max}
 
             $master client kill type slave
             set killed_used [s -1 used_memory]
             set killed_slave_buf [s -1 mem_clients_slaves]
             set killed_mem_not_counted_for_evict [s -1 mem_not_counted_for_evict]
             set killed_used_no_repl [expr {$killed_used - $killed_mem_not_counted_for_evict}]
-            set delta_no_repl [expr {$killed_used_no_repl - $used_no_repl}]
+            set delta_no_repl [expr {$killed_used_no_repl - $used_no_repl + $slave_qbuf}]
```

Comment:
i think you better convert the block that extracts the size of the query buffer to be a function so that you can run it twice.
we extract the memory usage details before and after the kill, and in this line we calculate the difference.
and we deduct the "repl" from both so that we only see the memory usage delta without the "repl".
but the thing is (what you discovered) is that we in addition to deducting the "repl" (i.e. `mem_not_counted_for_evict`),
we also need to deduct the slave query buffer (due to the ACKs inside it that are not in the control of that test).
so i think we need to deduct that size from both measurements (`killed_used_no_repl`, `used_no_repl`) rather than change the line you changed here.
and also add a comment as to why we're deducting it (due to the ACKs).